src/main/java/Storage.java

```diff
@@ -0,0 +1,159 @@
+                String taskType = taskDetails[0];
+                switch (taskType) {
+                    case "T":
+                        // Create new Task of todo type
+                        Task todo = new Task(Duke.Tasks.todo, taskDetails[2]);
+
+                        // Check if Task has been marked as done.
+                        if (Integer.parseInt(taskDetails[1]) == 1) {
+                            todo.done();
+                        }
+
+                        // Add todo to list
+                        tasks.add(todo);
+
+                        break;
+                    case "D":
+                        // Check if details provided in safe file is sufficient to form deadline Task
+                        if (taskDetails.length < 4) {
+                            throw new DukeException("insufficient details in deadline task :(");
+                        }
+
+                        // Create new Task of deadline type
+                        Task deadline = new Task(Duke.Tasks.deadline, taskDetails[2],
+                                taskDetails[3]);
+
+                        // Check if Task has been marked as done.
+                        if (Integer.parseInt(taskDetails[1]) == 1) {
+                            deadline.done();
+                        }
+
+                        // Add deadline to list.
+                        tasks.add(deadline);
+
+                        break;
+                    case "E":
+                        // Check if sufficient details are present to form Task of event type.
+                        if (taskDetails.length < 4) {
+                            throw new DukeException("insufficient details in event task :(");
+                        }
+
+                        // Create new Task object of event task type.
+                        Task event = new Task(Duke.Tasks.event, taskDetails[2], taskDetails[3]);
+
+                        // Check if Task has been marked as done.
+                        if (Integer.parseInt(taskDetails[1]) == 1) {
+                            event.done();
+                        }
+
+                        // Add event to list.
+                        tasks.add(event);
+
+                        break;
+                    default:
+                }
+            }
+        } catch (IOException ie) {
+            ie.printStackTrace();
+        }
+        return this.tasks;
+    }
+
+    /**
+     * Constructs a new list to store search results.
+     * @param searchKey Identifier that Task details needs to have.
+     * @return Task objects that contain identifier.
+     */
+    protected List<Task> find(String searchKey) {
+        // Create new List<Task> to hold search results.
+        List<Task> results = new ArrayList<>();
+        for (Task t: tasks) {
+            if (t.getDescription().contains(searchKey)) {
+                results.add(t);
+            }
+        }
+        return results;
+    }
+}
```

Comment:
I like that you tried to make it easier to read by adding line spacing and the comments you add for each part makes it easier to follow along.  As for line 72, I think javadoc return 4 spacing indentation. And just a suggestion, but maybe you can add find to the switch statement in the start method in Ui.